### ISSUE TITLE
Fairseq LM serialization outputTokensDim 

### ISSUE THREAD

**Issue by cri5Castro** [facebookresearch/wav2letter#506]
hello there After training a LM with fairseq I'm following the lexicon free serialization example, but i'm a little bit confused about the SerializeConvLM arguments, I understand that for word lm  the outputTokensDim is equal to 221452, I imagine that is the token dict dimension from dict.txt  but im a little bit confused with the char level lm which needs 40 outputTokensDim.

I'm using a word  piece tokenizer in my case the total amount of tokens are 9996 on my dict.txt from the  fairseq training data without considering the <fairseq_style>, <pad>, </s>, <unk> especial tokens so my  outputTokensDim for the Serializer should be  9996+4?

**Comment by tlikhomanenko** [comment]
Hi @cri5Castro,

Yep, for word level lm I had 221451 (221448 + 4 for special tokens) words in softmax layer output. For char lm we have 26 letters + 4 special tokens + {1, 2, ', |} for asg, and apostrophe, and silence and then 6 fake tokens because in fairseq they padded the number of output tokens to be divisible on 8 (but you can control this padding). So outputTokensDim should be len of generated `dict.txt` + 4
